Issue#250_Fix_exception_message_in_polls

kizimov opened the pull request:
dev
## ZenHub

* [Main ZenHub ticket](https://app.zenhub.com/workspaces/home-project-5f7b77ff8db73a001cb17009)


## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

kizimov commented:
@Disabled("Associated with Issue# 250. Will be fixed in issue # 290.")
CooperationContactApiIT.passNullUserIdWhenDeleteAnyContactTest
CooperationContactApiIT.passNullCooperationIdWhenGetContactTest
CooperationContactApiIT.passNullIdWhenDeleteAnyContactTest
UserApiIT.passNullWhenDeleteUserTest
NewsApiIT.passNullWhenDeleteNewsTest
CooperationContactApiIT.passNullIdWhenDeleteAnyContactTest


@Disabled("NotValid test. Will be fixed in issue # 290.")
ContactApiIT.passNullWhenGetContactTest (124 in Api.mustache)
UserApiIT.passNullWhenReceivingTest
NewsApiIT.passNullWhenReceivingNewsTest

@Disabled("error similar to Issue#250. Will be fixed in issue # 290.")
UserApiIT.createNullUserTest

@Disabled("broken test disabled")
QueryHousePollIT.getAllHousesFromPollsByQuantityFlat
QueryPoll. getAllPollsDescSort

kh0ma commented on home-api-tests/pom.xml:
```diff
@@ -401,5 +401,16 @@
         </dependency>
+        <dependency>
+            <groupId>org.assertj</groupId>
+            <artifactId>assertj-core</artifactId>
```
Isn't it redundant?

kizimov replied:
You are right. This is already in lines 383-386.  But then I need to delete line  385 (<version>${assertj-version}</version>)

kh0ma commented on home-api-tests/pom.xml:
```diff
@@ -401,5 +401,16 @@
+        </dependency>
+        <dependency>
+            <groupId>org.jetbrains</groupId>
+            <artifactId>annotations</artifactId>
```
Do we really need it?

kizimov replied:
We don't need this.

kh0ma commented on home-api-tests/src/test/java/com/softserveinc/ita/homeproject/api/tests/contacts/ContactApiIT.java:
```diff
@@ -43,8 +43,26 @@
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
```
Aren't these imports unused?

kizimov replied:
Removed unused imports.

kh0ma commented on home-api-tests/src/test/java/com/softserveinc/ita/homeproject/api/tests/invitations/InvitationApiIT.java:
```diff
@@ -12,12 +12,14 @@
 
     private final CooperationApi cooperationApi = new CooperationApi(ApiClientUtil.getClient());
 
+    @Disabled("NotValid test")
```
Please add more context why this test is not valid.

kh0ma replied:
It looks like it should be removed

kizimov replied:
You're right. I delete it.

kizimov committed:
Issue#250: Added api.mustache

kizimov committed:
Issue#250: api.mustache changed.

kh0ma commented on home-api-tests/src/test/java/com/softserveinc/ita/homeproject/api/tests/polls/QueryPoll.java:
```diff
@@ -70,7 +70,6 @@ void getAllPollsAscSort() throws ApiException {
     void getAllPollsDescSort() throws ApiException {
 
         createPoll();
-        createPoll();
```
Why was it removed?

kizimov replied:
I decided that in the previous tests we are already creating a number of polls. And that's enough. Maybe I was wrong.

kh0ma committed:
Issue#250 fix exception message in polls (#294)

* Issue#250: ValidationConfigurationContextResolver added to set annotation names for ConstraintViolationException.

* Issue#250: Page size integration test fixed

* Issue#250: Disabled test unblocked.

* Issue#250: Test is Disabled. A new issue will be initiated on it

* Issue#250: Custom mustache file added to avoid param check.

* Issue#250: First part of tests fixed.

* Issue#250: Added stub for test work.

* Issue#250: Code refactored to move provider and resolver to separate classes.

* Issue#250: New line to pass checkstyle added.

* Issue#250: Mustache file changed according to a new generator version.

* Issue#250 - Dropwizard implementation of ParameterNameProvider

* Issue#250: Aleksandr's suggestion merged.

* Issue#250: Logic fixed to pass IT.

* Issue#250: Deprecation class replaced.

* Issue#250: Unit tests added.

* Issue#250: Pom updated.

* Issue#250: Pom updated. Removed tests for JerseyParameterNameProvider.java.

* Issue#250: added new method in ApiClientUtil.java

* Issue#250: Changes were made to api.mustache. Disabled nullPointerException.

* Issue#250: Broken tests disabled.

* Issue#250: Broken tests disabled.

* Issue#250: Broken tests disabled.

* Issue#250: @Disabled annotation comment fix.

* Issue#250: Checked the tests and removed the annotation(@Disabled) in:
home-api-tests/src/test/java/com/softserveinc/ita/homeproject/api/tests/polls/QueryPoll.java;
home-api-tests/src/test/java/com/softserveinc/ita/homeproject/api/tests/polls/QueryHousePollIT.java;
home-api-tests/src/test/java/com/softserveinc/ita/homeproject/api/tests/invitations/InvitationApiIT.java

* Issue#250: Removed unused imports.

* Issue#250: Removed disabled dependency.

* Issue#250: Removed api.mustache

* Issue#250: Added api.mustache

* Issue#250: api.mustache changed.

* Issue#250: Unused imports removed.

* Issue#250: added createPoll() for getAllPollsDescSort();

Co-authored-by: Oleksiy-ver1 <kizimov1979@gmail.com>
Co-authored-by: Aleksey Kizimov <35989389+kizimov@users.noreply.github.com>
